packages/settings/setters/selectors/currencySelector.js

```diff
@@ -0,0 +1,34 @@
+import React, { useMemo } from 'react';
+import { toast } from 'react-toastify';
+import Select from '@basics/inputs/select';
+import settingConstants from '@settings/configuration/settingConstants';
+import useSettings from '@settings/managers/useSettings';
+import Piwik from '@common/utilities/piwik';
+
+function CurrencySelector({ t }) {
+  const { currency, toggleSetting } = useSettings(
+    settingConstants.keys.currency,
+  );
+  const onChangeCurrency = (value) => {
+    Piwik.trackingEvent('Settings', 'button', 'Update settings');
+    toggleSetting(value);
+    toast(t('Settings saved!'));
+  };
+
+  const currencies = useMemo(() => settingConstants.currencies.map((currencyName) => ({
+    label: currencyName,
+    value: currencyName,
+  })), [settingConstants.currencies]);
```

Comment:
static variable dosnt need to add as dependencies
https://stackoverflow.com/questions/64561538/why-does-passing-a-static-dependency-to-an-useeffect-causes-it-to-be-called-ever
```suggestion
  const currencies = useMemo(() => settingConstants.currencies.map((currencyName) => ({
    label: currencyName,
    value: currencyName,
  })), []);
```